File: Source/Timeline.zig
const db = @import("./Database.zig");
const utils = @import("./utils.zig");
pub const Day = u32;
const currentDay: ?Day = null;

pub const Timeline = struct {
    Id: db.Id,
    CurrentDay: Day,
};

pub fn GetCurrentDay() !Day {
    if (currentDay == null) {
        const timeline = try db.Session.query(Timeline).findFirst() orelse return utils.Error.Default;
        currentDay = timeline.CurrentDay;
    }
    return currentDay;
}

File: internal/timeline/timeline.go
package timeline

import (
	"worldboxing/lib/database"
	"worldboxing/lib/utils"
)

type Day = int
type Timeline struct {
	Id         utils.Id
	CurrentDay Day
}

var cachedCurrentDay *Day = nil

func CurrentDay() (Day, *utils.Error) {
	if cachedCurrentDay == nil {
		var timeline Timeline
		be := database.T.Select(&timeline, "SELECT * FROM Timeline LIMIT 1")
		if be != nil {
			panic("Unstarted world.")
		}
		cachedCurrentDay = &timeline.CurrentDay
	}
	return *cachedCurrentDay, nil
}
